authority/provisioner/azure.go

```diff
@@ -263,11 +263,19 @@ func (p *Azure) authorizeToken(token string) (*azurePayload, string, string, str
 	}
 
 	re := azureXMSMirIDRegExp.FindStringSubmatch(claims.XMSMirID)
-	if len(re) != 4 {
+	if len(re) != 5 {
 		return nil, "", "", "", "", errs.Unauthorized("azure.authorizeToken; error parsing xms_mirid claim - %s", claims.XMSMirID)
 	}
+
+	var subscription, group, name string
 	identityObjectID := claims.ObjectID
-	subscription, group, name := re[1], re[2], re[3]
+
+	if strings.Contains(claims.XMSMirID, "virtualMachines") {
+		subscription, group, name = re[1], re[2], re[4]
+	} else {
+		// This is not a VM resource ID so we don't have the VM name so set that to the empty string
+		subscription, group, name = re[1], re[2], ""
```

Comment:
I don't see the point of setting `name` to empty string here. That `name` will be set and validated as `commonName` only if `disableCustomSANs` is set to true. 

In fact, you mention that with disableCustomSANs enabled, it won't issue certs. I don't know what are the values on userAssignedIdentities, but after looking at examples it looks like its something that you can define. So I think that just removing that if condition and just using 
```
subscription, group, name = re[1], re[2], re[4]
```

Will work "ok" for both cases, when disableCustomSANs are disabled (default) or when they are enabled.